File: common/gateway-events.ts
export enum GameRoomEvents {
    JoinGame = 'joinGame',
    LeaveGame = 'leaveGame',
    JoinAccepted = 'joinAccepted',
    JoinDenied = 'joinDenied',
    AvatarUpdate = 'avatarUpdate',
    RoomUpdate = 'roomUpdateResponse',
    KickPlayer = 'kickPlayer',
    ToggleLock = 'toggleLock',
    ToggleFriendOnly = 'toggleFriendOnly',
    ToggleDropInDropOut = 'toggleDropInDropOut',
    SetLobbyGameMode = 'setLobbyGameMode',
    AddVirtualPlayer = 'addVirtualPlayer',
    KickUpdate = 'kickUpdate',
    StartGame = 'startGame',
    UpdateTeams = 'UpdateTeams',
    SelectTeam = 'SelectTeam',
    LeaveTeam = 'LeaveTeam',
    VirtualPlayerTeamChanged = 'VirtualPlayerTeamChanged',
    InsufficientCurrency = 'insufficientCurrency',
    ToggleFogOfWar = 'toggleFogOfWar',
    PublicRoomsUpdate = 'publicRoomsUpdate',
}

export enum ActiveGameEvents {
    CombatStarted = 'combatStarted',
    CombatInitiated = 'combatInitiated',
    CombatAction = 'combatAction',
    CombatUpdate = 'combatUpdate',
    MovePlayer = 'movePlayer',
    PlayerNextPosition = 'playerNextPosition',
    PlayerStartedMoving = 'playerStartedMoving',
    PlayerStoppedMoving = 'PlayerStoppedMoving',
    TurnUpdate = 'turnUpdate',
    NextTurn = 'nextTurn',
    NoMorePlayers = 'noMorePlayers',
    PlayerDisconnect = 'playerDisconnect',
    ToggledDoor = 'toggledDoor',
    DoorUpdate = 'doorUpdated',
    FetchStats = 'fetchStats',
    GameEnded = 'gameEnded',
    MapRequest = 'mapRequest',
    ItemSwapped = 'itemSwapped',
    ItemUpdate = 'itemUpdate',
    ItemsDropped = 'itemsDropped',
    ResetInventory = 'resetInventory',
    ItemPickedUp = 'itemPickedUp',
    TradeInit = 'TradeInit',
    TradeStarted = 'TradeStarted',
    TradeUpdate = 'TradeUpdate',
    TradeAccept = 'TradeAccept',
    TradeCancel = 'TradeCancel',
    TradeComplete = 'TradeComplete',
    DropIn = 'dropIn',
    SpawnPlayer = 'spawnPlayer',
    SelectCombatTarget = 'SelectCombatTarget',
    PlayerReady = 'playerReady',
}

export enum GameChatEvents {
    SendMessage = 'sendMessage',
    ReceiveMessage = 'receiveMessage',
    ReceiveTeamMessage = 'receiveTeamMessage',
}

export enum GlobalChannelEvents {
    SendMessage = 'globalChannel:sendMessage',
    ReceiveMessage = 'globalChannel:receiveMessage',
    AccountDeleted = 'globalChannel:accountDeleted',
}

export enum CustomChannelEvents {
    SendMessage = 'customChannel:sendMessage',
    ReceiveMessage = 'customChannel:receiveMessage',
    RetrieveMessages = 'customChannel:retrieveMessages',
    GiveMessages = 'customChannel:giveMessages',
    Create = 'customChannel:create',
    Join = 'customChannel:join',
    Leave = 'customChannel:leave',
    Search = 'customChannel:search',
    SearchResults = 'customChannel:searchResults',
    ChannelDeleted = 'customChannel:deleted',
    CloseChannel = 'customChannel:close',
    GetJoinedChannels = 'customChannel:getJoined',
    GiveJoinedChannels = 'customChannel:giveJoined',
    Error = 'customChannel:error',
    newOwner = 'customChannel:newOwner',
}

export enum TimerEvents {
    StartTimer = 'start-timer',
    StopTimer = 'stop-timer',
    TimerUpdate = 'timer-update',
    TimerEnd = 'timer-end',
    ResetTimer = 'reset-timer',
}

export enum DebugEvents {
    ToggleDebug = 'debug',
    RequestDebugState = 'requestDebugState',
}

export enum CTFEvents {
    FlagTaken = 'flagTaken',
    FlagDropped = 'flagDropped',
    FlagCaptured = 'flagCaptured',
}

export enum VirtualCurrencyEvents {
    CurrencyUpdate = 'currencyUpdate',
}

File: mobile_client/lib/app/gateway_events.dart
/// Aligned with `common/gateway-events.ts` (Socket.IO events used by the mobile client).
abstract final class GameRoomSocketEvents {
  static const joinGame = 'joinGame';
  static const leaveGame = 'leaveGame';
  static const joinAccepted = 'joinAccepted';
  static const joinDenied = 'joinDenied';
  static const avatarUpdate = 'avatarUpdate';
  static const roomUpdateResponse = 'roomUpdateResponse';
  static const kickPlayer = 'kickPlayer';
  static const toggleLock = 'toggleLock';
  static const toggleFriendOnly = 'toggleFriendOnly';
  static const toggleDropInDropOut = 'toggleDropInDropOut';
  static const setLobbyGameMode = 'setLobbyGameMode';
  static const addVirtualPlayer = 'addVirtualPlayer';
  static const kickUpdate = 'kickUpdate';
  static const startGame = 'startGame';
  static const updateTeams = 'UpdateTeams';
  static const selectTeam = 'SelectTeam';
  static const leaveTeam = 'LeaveTeam';
  static const virtualPlayerTeamChanged = 'VirtualPlayerTeamChanged';
  static const toggleFogOfWar = 'toggleFogOfWar';
  static const publicRoomsUpdate = 'publicRoomsUpdate';
}

abstract final class VirtualCurrencySocketEvents {
  static const currencyUpdate = 'currencyUpdate';
}

/// Aligned with `GlobalChannelEvents` in `common/gateway-events.ts`.
abstract final class GlobalChannelSocketEvents {
  static const accountDeleted = 'globalChannel:accountDeleted';
}

const deletedAccountUsername = '[deleted_account]';

abstract final class GameChatSocketEvents {
  static const sendMessage = 'sendMessage';
  static const receiveMessage = 'receiveMessage';
  static const receiveTeamMessage = 'receiveTeamMessage';
}

/// Aligned with `CustomChannelEvents` in `common/gateway-events.ts`.
abstract final class CustomChannelSocketEvents {
  static const sendMessage = 'customChannel:sendMessage';
  static const receiveMessage = 'customChannel:receiveMessage';
  static const retrieveMessages = 'customChannel:retrieveMessages';
  static const giveMessages = 'customChannel:giveMessages';
  static const create = 'customChannel:create';
  static const join = 'customChannel:join';
  static const leave = 'customChannel:leave';
  static const search = 'customChannel:search';
  static const searchResults = 'customChannel:searchResults';
  static const channelDeleted = 'customChannel:deleted';
  static const closeChannel = 'customChannel:close';
  static const getJoinedChannels = 'customChannel:getJoined';
  static const giveJoinedChannels = 'customChannel:giveJoined';
  static const newOwner = 'customChannel:newOwner';
  static const error = 'customChannel:error';
}
